app/controllers/sks_users_controller.ts

```diff
@@ -0,0 +1,93 @@
+import { HttpContext } from '@adonisjs/core/http'
+import SksUser from '#models/sks_user'
+import { DateTime } from 'luxon'
+
+//this value determines the period over which the trend will be counted (multiply the value by 5 minutes)
+const trendDelta = 3
+
+enum Trend {
+  INCREASING = 'INCREASING',
+  DECREASING = 'DECREASING',
+  STABLE = 'STABLE',
+}
+
+export default class SksUsersController {
+  /**
+   * Display the latest SKS users record based on time
+   */
+  async latest({ response }: HttpContext) {
+    try {
+      const currentTime = DateTime.now().setZone('Europe/Warsaw').toSQL()
+      if (currentTime === null) {
+        return response.status(500).json({ message: 'Failed to convert time to sql format' })
+      }
+
+      const latestData = await SksUser.query()
+        .where('externalTimestamp', '<', currentTime)
+        .orderBy('externalTimestamp', 'desc')
+        .offset(1) //get the second-latest record to ensure the data is already scraped
+        .first()
+      if (latestData === null) {
+        return response
+          .status(404)
+          .json({ message: 'Could not find the matching data in database' })
+      }
+
+      let trend: Trend
+      const trendData = await SksUser.query()
+        .where('externalTimestamp', '<', currentTime)
+        .orderBy('externalTimestamp', 'desc')
+        .offset(1 + trendDelta)
+        .first()
+      if (trendData === null) {
+        trend = Trend.STABLE // If no previous data, assume stable trend
+      } else {
+        const isIncreasing = trendData.activeUsers < latestData.activeUsers
+        const isDecreasing = trendData.activeUsers > latestData.activeUsers
+
+        if (isIncreasing) {
```

Comment:
what you've done is a really nice practice (extracting boolean variables that have descriptive names). I really like it, but is it really needed in this simple case 😭

Reply:
but I can't complain, you can leave it - the code here is just too good to find other problems so I decided to share a thought

Reply:
okay, the coding marathon hit hard on this one 😭
gonna leave it, cause it does not hurt though
also this is the way of Przewo

Reply:
I know this is the way of Przewo

Reply:
And it's generally a nice practise

Reply:
(when you have complex conditions 😭 )